docs/src/main/asciidoc/kafka-schema-registry-avro.adoc

```diff
@@ -0,0 +1,604 @@
+mp.messaging.outgoing.movies.topic=movies
+
+# set the serializer for the `movies` channel to the Avro Serializer for Apicurio
+mp.messaging.outgoing.movies.value.serializer=io.apicurio.registry.utils.serde.AvroKafkaSerializer
+
+# Apicurio schema specific settings:
+mp.messaging.outgoing.movies.apicurio.registry.artifact-id=io.apicurio.registry.utils.serde.strategy.SimpleTopicIdStrategy
+mp.messaging.outgoing.movies.apicurio.registry.global-id=io.apicurio.registry.utils.serde.strategy.GetOrCreateIdStrategy
+mp.messaging.outgoing.movies.apicurio.registry.avro-datum-provider=io.apicurio.registry.utils.serde.avro.ReflectAvroDatumProvider
+----
+
+== The `Movie` consumer
+So, we can write records into Kafka containing our `Movie` data.
+That data is serialized using Avro.
+Now, it's time to implement a consumer for them.
+
+Let's create `ConsumedMovieResource` that will consume `Movie` messages
+from the `movies-from-kafka` channel and will expose it via Server-Sent Events:
+
+[source,java]
+----
+package org.acme.kafka;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.acme.kafka.quarkus.Movie;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.jboss.resteasy.annotations.SseElementType;
+
+import io.smallrye.mutiny.Multi;
+
+@ApplicationScoped
+@Path("/consumed-movies")
+public class ConsumedMovieResource {
+
+    @Channel("movies-from-kafka")
+    Multi<Movie> movies;
+
+    @GET
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    @SseElementType(MediaType.TEXT_PLAIN)
+    public Multi<String> stream() {
+        return movies.map(movie -> String.format("'%s' from %s", movie.getTitle(), movie.getYear()));
+    }
+}
+----
+
+The last bit of the application's code is the configuration of the `movies-from-kafka` channel in
+`application.properties`:
+[source,properties]
+----
+# set the connector for the incoming channel to `smallrye-kafka`
+mp.messaging.incoming.movies-from-kafka.connector=smallrye-kafka
+
+# set the topic name for the channel to `movies`
+mp.messaging.incoming.movies-from-kafka.topic=movies
+
+# set the deserializer for the `movies-from-kafka` channel to the Avro Deserializer for Apicurio
+mp.messaging.incoming.movies-from-kafka.value.deserializer=io.apicurio.registry.utils.serde.AvroKafkaDeserializer
+
+# disable auto-commit, Reactive Messaging handles it itself
+mp.messaging.incoming.movies-from-kafka.enable.auto.commit=false
+
+mp.messaging.incoming.movies-from-kafka.auto.offset.reset=earliest
+mp.messaging.incoming.movies-from-kafka.apicurio.registry.avro-datum-provider=io.apicurio.registry.utils.serde.avro.ReflectAvroDatumProvider
+----
+
+
+== The infrastructure
+To use our application, we need Kafka and Apicurio Schema Registry.
+The easiest way to get them running is to use `docker-compose` to start the appropriate containers.
```

Comment:
Here is a tricky part... `docker-compose` became `docker compose` in the last docker version. I would keep it like this for now, but we may have to update (globally) the reference.